### PR TITLE
Rewrite role to typescript

### PR DESCRIPTION
**Pull request by miherlosev**
Changes:
* rewrite Role related files to TypeScript
* use `loginUrl` and `redirectUrl` terms into Role
* add `Role.setCurrentUrlAsRedirectUrl()` method.

### CODE REVIEW

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
FPR

File: src/api/test-page-url.ts [hunk 1]
```diff
@@ -34,12 +34,12 @@ export function assertUrl (url: string, callsiteName: string): void {
         throw new APIError(callsiteName, RUNTIME_ERRORS.unsupportedUrlProtocol, url, protocol && protocol[0]);
 }
 
-export function resolvePageUrl (url: string, testFileName: string): string {
+export function resolvePageUrl (url: string, testFileName?: string): string {
```
Comment (AndreyBelym) [inline src/api/test-page-url.ts:37]: The `testFileName` is actually a required argument. When it is not passed, `resolvePageUrl` can throw unexpected errors:
https://github.com/DevExpress/testcafe/issues/3406#issuecomment-514834132
https://stackoverflow.com/questions/60527374/testcafe-throws-an-error-invalid-arg-type

Using `testFileName as string` to suppress type checking errors is dangerous, we should fix the problem by improving the resolution algorithm or by detecting the problem and throwing a descriptive error.

**Review by AndreyBelym — CHANGES_REQUESTED** [review]

**Comment by miherlosev** [comment]
FPR

**Review by AndreyBelym — APPROVED** [review]

**Review by wentwrong — APPROVED** [review]

**Pull request merged**